osu.Server.Queues.ScoreStatisticsProcessor/Commands/Performance/UserTotals/UpdateUserTotalsFromSqlCommand.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dapper;
+using JetBrains.Annotations;
+using McMaster.Extensions.CommandLineUtils;
+using osu.Server.QueueProcessor;
+using osu.Server.Queues.ScoreStatisticsProcessor.Helpers;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Commands.Performance.UserTotals
+{
+    [Command("sql", Description = "Updates the total PP of scoped users.")]
+    public class UpdateUserTotalsFromSqlCommand : PerformanceCommand
+    {
+        [UsedImplicitly]
+        [Required]
+        [Argument(0, Description = "Specify a custom query to limit the scope of users to process.")]
+        public string? Query { get; set; }
```

Comment:
I can't *not* ask... is there a reason for allowing arbitrary sql constructions to be part of the query? Is there nothing more... constrained that I can be offered here? I'm too worried about bobby tables shenanigans to *not* ask this question...

Reply:
Reason is flexibility and keeping things simple from a code perspective. See sample query which may and will involve `WHERE`, `JOIN`s, `LIMIT`s and `ORDER`s.